modules/vye/app/models/concerns/vye/gen_digest.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'openssl'
+
+module Vye; end
+
+module Vye::GenDigest
+  extend ActiveSupport::Concern
+
+  module Common
+    GEN_DIGEST_CONFIG =
```

Comment:
I have a hard time believing this functionality isn't already in something like `Digest::SHA256`. Is there a reason you're doing this custom here?

Reply:
For example, in this file, `app/services/sign_in/session_creator.rb`, we do a one-way hash of an object for later comparison with `Digest::SHA256.hexdigest(object)`

Reply:
Absolutely, I understand your concerns around the usage of a custom algorithm instead of a well-known one like `Digest::SHA256`. However, the decision to utilize scrypt over SHA256 is primarily grounded in security considerations.

The SHA256 algorithm, while reliable, is susceptible to brute force attacks. This is because it was not engineered to be slow or to inhibit the usage of a GPU, which can significantly speed up the process of cracking the hash.

On the other hand, scrypt was explicitly designed to address these vulnerabilities. It's resistant to brute force attacks by GPUs, and its speed can be deliberately reduced with the help of "work factor" parameters (N, r, p, and length). This effectively increases the time and computational power required to crack the hash.

If we were to use SHA256 in our scenario and someone obtains a database dump, they could potentially brute force the salt if they know one of the Social Security Numbers (SSNs). Once the salt is compromised, it becomes possible to brute force the remaining SSNs.

Scrypt, therefore, provides us with an additional layer of security against such attacks, making it a more suitable choice for our needs.

https://en.wikipedia.org/wiki/Scrypt

Reply:
@bosawt as per our phone conversation just now I will continue to use the scrypt function from the OpenSSL gem as it is in the PR now without modifications.

Reply:
I was thinking about this, why not sha hash and then encrypt it? Then it would be safe from attackers and you wouldn't have to deal with your own seed values on each stack you could just leverage the existing encryption.

Reply:
I appreciate your suggestion about using the SHA256 hash and then encrypting it. However, it's important to note that encryption isn't viable in this scenario because the column needs to be indexed for fast lookups. Furthermore, regardless of whether we use SHA256 or another method, salting per environment would still be necessary without encryption.